src/pydata_sphinx_theme/theme/pydata_sphinx_theme/layout.html

```diff
@@ -96,6 +81,10 @@
   <div class="bd-container">
     <div class="bd-container__inner bd-page-width">
       {# Primary sidebar #}
+      {# If we have no sidebar TOC, pop the TOC component from the sidebar list #}
+      {% if get_sidebar_toctree_length(show_nav_level=theme_show_nav_level|int) == 0 %}
+        {% set sidebars = sidebars | reject("in", "sidebar-nav-bs.html") | list %}
+      {% endif %}
```

Comment:
here's where that conditional ended up. It now calls `get_sidebar_toctree_length` which looks at the *unrendered* toctree